[rpc] move get_rpc_timeout to pybind

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29765 [rpc] move get_rpc_timeout to pybind**

instead of wrapping this C++ function with python that causes
unnecessary overhead, we can move this to pybind and use the `DefaultRpcAgent`
to get the timeout. This was @pritamdamania87's suggestion in the comments on this PR: https://github.com/pytorch/pytorch/pull/29341/files#r345003483

Differential Revision: [D18493195](https://our.internmc.facebook.com/intern/diff/D18493195/)

rohan-varma committed:
[rpc] move get_rpc_timeout to pybind

instead of wrapping this C++ function with python that causes
unnecessary overhead, we can move this to pybind and use the `DefaultRpcAgent`
to get the timeout.

Differential Revision: [D18493195](https://our.internmc.facebook.com/intern/diff/D18493195/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] move get_rpc_timeout to pybind"


instead of wrapping this C++ function with python that causes
unnecessary overhead, we can move this to pybind and use the `DefaultRpcAgent`
to get the timeout. This was @pritamdamania87's suggestion in the comments on this PR: https://github.com/pytorch/pytorch/pull/29341/files#r345003483

Differential Revision: [D18493195](https://our.internmc.facebook.com/intern/diff/D18493195/)

[ghstack-poisoned]

pritamdamania87 commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -168,6 +166,10 @@ PyObject* rpc_init(PyObject* /* unused */) {
 
+  module.def("get_rpc_timeout", []() {
+    return RpcAgent::getDefaultRpcAgent()->getRpcTimeout();
+  });
```
We should add the docstring here.

rohan-varma replied:
Added

rohan-varma committed:
Update on "[rpc] move get_rpc_timeout to pybind"


instead of wrapping this C++ function with python that causes
unnecessary overhead, we can move this to pybind and use the `DefaultRpcAgent`
to get the timeout. This was @pritamdamania87's suggestion in the comments on this PR: https://github.com/pytorch/pytorch/pull/29341/files#r345003483

Differential Revision: [D18493195](https://our.internmc.facebook.com/intern/diff/D18493195/)

[ghstack-poisoned]

rohan-varma commented on torch/distributed/rpc/__init__.py:
```diff
@@ -53,7 +53,7 @@ def init_model_parallel(
             init_method(str): backend specific init arguments.
             num_send_recv_threads(int): Number of threads for send/recv work.
-            rpc_timeout (datetime.timedelta): Timeout for RPCs. Defaults to 10 seconds.
+            rpc_timeout (datetime.timedelta): Timeout for RPCs. Defaults to 60 seconds.
```
This was incorrect as we decided to change the default timeout and this was not updated.

rohan-varma committed:
Update on "[rpc] move get_rpc_timeout to pybind"


instead of wrapping this C++ function with python that causes
unnecessary overhead, we can move this to pybind and use the `DefaultRpcAgent`
to get the timeout. This was @pritamdamania87's suggestion in the comments on this PR: https://github.com/pytorch/pytorch/pull/29341/files#r345003483

Differential Revision: [D18493195](https://our.internmc.facebook.com/intern/diff/D18493195/)

[ghstack-poisoned]

xush6528 commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -168,6 +166,16 @@ PyObject* rpc_init(PyObject* /* unused */) {
+      "get_rpc_timeout",
+      []() { return RpcAgent::getDefaultRpcAgent()->getRpcTimeout(); },
+      R"(
+          Retrieve the timeout for all RPCs that was set during RPC initialization.
```
I am guessing this will bring s lot of spaces in the front?

Should we start from the beginning of a line?

https://github.com/pytorch/pytorch/blob/3bc014ecf2a1badee5b1aa19e8b99273134f9590/caffe2/sgd/learning_rate_op.cc#L16

rohan-varma replied:
@xush6528 it looks like it is autoformatted, if you build this PR and do `import torch.distributed.rpc as rpc ; help(rpc.get_rpc_timeout)` the format looks fine. Though I will check how this appears on the docs.
